backends/ebpf/bpfinclude/ebpf_registry.c

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2018 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Implementation of ebpf registry. Intended to provide a common access interface between control and data plane. Emulates the linux userspace API which can access the kernel eBPF map using string and integer identifiers.
+*/
+
+#include <stdio.h>
+#include "ebpf_registry.h"
+
+/**
+ * @brief Defines the structure of the central registry.
+ * @details Defines a registry type, which maps names to tables
+ * as well as integer identifiers.
+ */
+typedef struct {
+    char name[MAX_TABLE_NAME_LENGTH];         // name of the map
+    int map_handle;                           // id of the map
+    struct bpf_map_def *table;                // ptr to the map
+    UT_hash_handle hh;                        // makes this structure hashable
+} registry_entry;
+
+/**
+ * @brief A mapping between names and handles.
+ * @details Maps the table names to their respective integer handles.
+ * Provides a mechanism to find a table name by its integer value
+ * representation.
+ */
+typedef struct {
+    int map_handle;                    // file descriptor of the map
+    char name[MAX_TABLE_NAME_LENGTH];  // name of the map
+    UT_hash_handle hh;                 // makes this structure hashable
+} handle_entry;
+
+
+static int map_indexer = 0;
+
+/* Instantiation of the central registry */
+registry_entry *reg_tables = NULL;
+handle_entry *table_handles = NULL;
+
+
+static registry_entry *find_table(const char *name) {
+    if (strlen(name) > MAX_TABLE_NAME_LENGTH){
+        fprintf(stderr, "%s", "Error: Key length exceeds maximum size!\n");
```

Comment:
I don't think that the '%s' is necessary here, but you could print the name.
Also, you may check if name is NULL and return an error.